src/app/zap-templates/zcl/data-model/chip/energy-evse-cluster.xml

```diff
@@ -0,0 +1,235 @@
+    <attribute side="server" code="0x0023" define="NEXT_CHARGE_START_TIME"      type="epoch_s"                    writable="false" default=""       optional="true"  isNullable="true">NextChargeStartTime</attribute>
+    <attribute side="server" code="0x0024" define="NEXT_CHARGE_TARGET_TIME"     type="epoch_s"                    writable="false" default=""       optional="true"  isNullable="true">NextChargeTargetTime</attribute>
+    <attribute side="server" code="0x0025" define="NEXT_CHARGE_REQUIRED_ENERGY" type="INT32S"                     writable="false" default=""       optional="true"  isNullable="true">NextChargeRequiredEnergy</attribute>
+    <attribute side="server" code="0x0026" define="NEXT_CHARGE_TARGET_SOC"      type="INT8U"                      writable="false" default=""       optional="true"  isNullable="true">NextChargeTargetSoc</attribute>
+    <attribute side="server" code="0x0027" define="APPROX_EV_EFFICIENCY"        type="INT16U"                     writable="true"  default=""       optional="true"  isNullable="true">ApproxEvEfficiency</attribute>
+    <attribute side="server" code="0x0030" define="STATE_OF_CHARGE"             type="Percent" min="0" max="100"  writable="false" default=""       optional="true"  isNullable="true">StateOfCharge</attribute>
+    <attribute side="server" code="0x0031" define="BATTERY_CAPACITY"            type="INT32S"                     writable="false" default=""       optional="true"  isNullable="true">BatteryCapacity</attribute>
+    <attribute side="server" code="0x0032" define="VEHICLE_ID"                  type="CHAR_STRING"                writable="false" default=""       optional="true"  isNullable="true" length="32" >VehicleId</attribute>
+
+    <attribute side="server" code="0x0040" define="SESSION_ID"                  type="INT32U"                     writable="false" default="0x0"    optional="true"  isNullable="false">SessionId</attribute>
+    <attribute side="server" code="0x0041" define="EVENT_SEQUENCE_NUMBER"       type="INT16U"                     writable="false" default="0x0001" optional="true"  isNullable="false">EventSequenceNumber</attribute>
+    <attribute side="server" code="0x0042" define="SESSION_DURATION"            type="elapsed_s"                  writable="false" default="0x0"    optional="true"  isNullable="false">SessionDuration</attribute>
+    <attribute side="server" code="0x0043" define="SESSION_ENERGY_CHARGED"      type="INT32S"                     writable="false" default="0x0"    optional="true"  isNullable="false">SessionEnergyCharged</attribute>
+    <attribute side="server" code="0x0044" define="SESSION_ENERGY_DISCHARGED"   type="INT32S"                     writable="false" default="0x0"    optional="true"  isNullable="false">SessionEnergyDischarged</attribute>
```

Comment:
all of these (some above too) say `INT32S` and the spec says `ref_Energy` ... and I cannot find the energy reference. I do not know if this is ok or not (e.g. can energy be negative?). Could you help me find this and validate?

Reply:
And in another place I mentioned, refEnergy seemed unsigned but here we have tons of signed. We need a def for this somewhere.

Reply:
Yep - thought there was a ref_energy defined in the types, but clearly not. Will add either to data model .adoc or here.
In terms of signed vs unsigned. Energy can flow in both directions, but in metering you tend to have an imported and a separate exported register so that you track the cumulative total import and total export separately. Most of the time I think it would be unsigned. But if we have a generic type then I think it should probably be signed, and where it is used could be specified as to what is allowed?